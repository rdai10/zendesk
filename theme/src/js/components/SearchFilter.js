import preact from 'preact';
import PropTypes from 'prop-types';

class SearchFilter extends preact.Component {
	constructor(props) {
		super(props);

		this.handleChange = this.handleChange.bind(this);
	}

	handleChange(event) {
		const {onChange} = this.props;

		onChange(event.currentTarget.value);
	}

	render({label, options}) {
		return (
			<div class="search-filter">
				<label class="semi-bold" htmlFor='productSearchFilter'>
					{label}
				</label>

				<select class="col-md-3 input-small" id="productSearchFilter" onChange={this.handleChange}>
					{options.map(
						(option, index) => (
							<option key={index} value={option.value}>
								{option.displayName}
							</option>
						)
					)}
				</select>
			</div>
		);
	}
}

SearchFilter.PropTypes = {
	label: PropTypes.string,
	onChange: PropTypes.func.isRequired,
	options: PropTypes.arrayOf(
		PropTypes.shape(
			{
				displayName: PropTypes.string,
				value: PropTypes.string
			}
		)
	).isRequired
};

export default SearchFilter;