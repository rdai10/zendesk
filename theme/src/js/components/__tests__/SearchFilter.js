import preact from 'preact';
import {cleanup, fireEvent, render} from 'preact-testing-library';

import SearchFilter from '../SearchFilter';

const setup = () => {
	const handleOnChange = jest.fn();

	const options = [
		{
			displayName: 'Option 1',
			value: 'one'
		},
		{
			displayName: 'Option 2',
			value: 'two'
		}
	];

	const utils = render(
		<SearchFilter
			label='Search Filter'
			onChange={handleOnChange}
			options={options}
		/>
	);

	return {
		handleOnChange,
		...utils
	};
};

afterEach(cleanup);

describe('SearchFilter', () => {
	it('renders correctly', () => {
		const {container} = setup();

		expect(container).toBeTruthy();
	});

	it('populates select options', () => {
		const {getByValue} = setup();

		const option1 = getByValue('one');
		const option2 = getByValue('two');

		expect(option1).toBeDefined();
		expect(option2).toBeDefined();
	});

	it('does something when a new option is selected', () => {
		const {container, handleOnChange} = setup();

		const filter = container.querySelector('select');

		fireEvent.change(filter, {target: {value: 'Option 2'}});

		expect(handleOnChange).toHaveBeenCalledTimes(1);
	});
});
