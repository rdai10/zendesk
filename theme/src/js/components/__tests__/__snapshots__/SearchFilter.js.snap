// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter renders correctly 1`] = `
<div>
  <div
    class="search-filter"
  >
    <label
      class="semi-bold"
      for="productSearchFilter"
    >
      Search Filter
    </label>
    <select
      class="col-md-3 input-small"
      id="productSearchFilter"
    >
      <option
        value="one"
      >
        Option 1
      </option>
      <option
        value="two"
      >
        Option 2
      </option>
    </select>
  </div>
</div>
`;
